client/allocrunner/taskrunner/connect_hook.go

```diff
@@ -0,0 +1,145 @@
+package taskrunner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	agentconsul "github.com/hashicorp/nomad/command/agent/consul"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+var _ interfaces.TaskPrestartHook = &connectHook{}
+
+// connectHook writes the bootstrap config for the Connect proxy sidecar
+type connectHook struct {
+	alloc          *structs.Allocation
+	consulHTTPAddr string
```

Comment:
Can you comment here how this is not actually used by the sidecar and just for building the bootstrap. I forgot and was about to ask why this isn't a unix socket either.

Reply:
Good call. The bootstrap vs runtime dependency question confused me for a while when implementing it, so I'll add comments.